lang/java/build.sh

```diff
@@ -35,6 +35,11 @@ main() {
         # Test the modules that depend on hadoop using Hadoop 2
         mvn -B test -Phadoop2
         ;;
+      testJava17)
+        mvn -DdisableJava17=false -B test
+        # Test the modules that depend on hadoop using Hadoop 2
+        mvn -B test -Phadoop2
```

Comment:
IMO we can execute just `mvn -DdisableJava17=false -B test` at https://github.com/apache/avro/pull/1680/commits/50977227b1500dd73763c99e7962a00f18d7ec6d#diff-a0e5c0a6a98abf33065a9bfa4dd160ca920278cbdbfabed88cb317a10df9f294R83 and remove `testJava17` from here

Reply:
Happy to make that change. Just want to put rationale before doing that. Might change your mind 

With the change to make the module included by default but skip the compile/tests.

running `mvn -DdisableJava17=false -B test` will run all the tests in every module. Can add more `mvn` args to try and narrow the rerun down to only what is needed. But figured in the main test run for java17+ enable the module preventing the need to redo all the other work. This way it only adds a few seconds to the build. As opposed to redoing the step that takes minutes.